utils/test_harness/tools/src/test_harness_metric.cpp

```diff
@@ -582,6 +582,18 @@ void metric_streamer_read_data(
   rawDataSize = metricSize;
 }
 
+void metric_streamer_read_data(
+    zet_metric_streamer_handle_t metricStreamerHandle, uint32_t reports,
+    size_t &rawDataSize, std::vector<uint8_t> *metricData) {
+  ASSERT_NE(nullptr, metricData);
+  rawDataSize = metric_streamer_read_data_size(metricStreamerHandle, reports);
+  EXPECT_GT(rawDataSize, 0);
+  metricData->resize(rawDataSize);
```

Comment:
Lets have all allocations in the test itself

Reply:
These tests are now not checking if event was generated, so if we allocate in test we might over allocate from what is really needed. Is this okay?

Reply:
The comment was to move the steps in this harness to the test itself.
Allocating under a harness makes the test functionality less obvious about the steps involved.

Reply:
done